DoLah/Models/Task.cpp

```diff
@@ -30,6 +30,17 @@ namespace DoLah {
         return this->tags;
     }
 
+    void AbstractTask::setExpired(bool expired) {
+        this->expired = expired;
+    }
+
+    bool AbstractTask::isExpired() {
+        return this->expired;
+    }
```

Comment:
This function should be calculate whether it is expiry at the time when it was called, and the value of expiration should be computed directly when calling this function.